x-pack/legacy/plugins/maps/public/layers/styles/vector/components/color/color_stops_categorical.js

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { EuiFieldText } from '@elastic/eui';
+import {
+  addCategoricalRow,
+  isCategoricalStopsInvalid,
+  getOtherCategoryLabel,
+  DEFAULT_CUSTOM_COLOR,
+  DEFAULT_NEXT_COLOR,
+} from './color_stops_utils';
+import { i18n } from '@kbn/i18n';
+import { ColorStops } from './color_stops';
+
+export const ColorStopsCategorical = ({
+  colorStops = [
+    { stop: null, color: DEFAULT_CUSTOM_COLOR }, //first stop is the "other" color
+    { stop: '', color: DEFAULT_NEXT_COLOR },
+  ],
+  onChange,
+}) => {
+  const sanitizeStopInput = value => {
+    return value;
+  };
+
+  const getStopError = (stop, index) => {
+    let count = 0;
+    for (let i = 1; i < colorStops.length; i++) {
+      if (colorStops[i].stop === stop && i !== index) {
+        count++;
+      }
+    }
+
+    return count
+      ? i18n.translate('xpack.maps.styles.colorStops.categoricalStop.noDupesWarningLabel', {
+          defaultMessage: 'Stop values must be unique',
+        })
+      : null;
+  };
+
+  const renderStopInput = (stop, onStopChange, index) => {
+    let stopInput;
+    if (index === 0) {
+      stopInput = (
```

Comment:
It would be more consistent with existing styling to just return component here and then remove else statement and `stopInput` variable